apps/api/src/app/topics/use-cases/rename-topic/rename-topic.use-case.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { TopicEntity, TopicRepository } from '@novu/dal';
+
+import { RenameTopicCommand } from './rename-topic.command';
+
+import { GetTopicUseCase, GetTopicCommand } from '../get-topic';
+import { TopicDto } from '../../dtos/topic.dto';
+
+@Injectable()
+export class RenameTopicUseCase {
+  constructor(private getTopicUseCase: GetTopicUseCase, private topicRepository: TopicRepository) {}
+
+  async execute(command: RenameTopicCommand): Promise<TopicDto> {
+    await this.getTopicUseCase.execute(command);
+
+    const entity = this.mapToEntity(command);
```

Comment:
A small note, i'm not sure that map to entity is a correct name here, as it more about the query for the dal and not the entity itself.